walletnode/services/senseregister/task.go

```diff
@@ -40,8 +46,123 @@ func (task *Task) Run(ctx context.Context) error {
 	log.WithContext(ctx).Debug("Start task")
 	defer log.WithContext(ctx).Debug("End task")
 
-	// TODO: implement task logic
+	if err := task.run(ctx); err != nil {
+		task.UpdateStatus(StatusTaskRejected)
+		log.WithContext(ctx).WithErrorStack(err).Error("Task is rejected")
+		return nil
+	}
 
 	task.UpdateStatus(StatusTaskCompleted)
 	return nil
 }
+
+func (task *Task) run(ctx context.Context) error {
+	_, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// Get top 10MNs from pastel network
+	topNodes, err := task.pastelTopNodes(ctx)
+	if err != nil {
+		return errors.Errorf("get top 10MNs from pastel network: %v", err)
+	}
+
+	if len(topNodes) < task.config.NumberSuperNodes {
+		return errors.New("unable to find enough Supernodes from pastel network")
+	}
+
+	// Get current block height & hash
+	if err := task.getBlock(ctx); err != nil {
+		return errors.Errorf("get block: %v", err)
+	}
+
+	// Connect to top nodes to validate their infor
+	err = task.validateMNsInfo(ctx, topNodes)
+	if err != nil {
+		return errors.Errorf("validate MNs info: %v", err)
+	}
+
+	return nil
+}
+
+// pastelTopNodes - get 10 top MNs from pastel network
+func (task *Task) pastelTopNodes(ctx context.Context) (node.List, error) {
```

Comment:
this code already exists... in artwork register I believe... can we use on all sides from a common place?